### ISSUE TITLE
Crashes from games

### ISSUE THREAD

**Issue by mhauskn** [Microsoft/jericho#17]
balances.z5 - "take her" --> Fatal Error: Illegal Object
detective.z5 - "showverb eight" --> hang; [** Programming error: tried to read from -->1115 in the array " [** Programming error: tried to print (string) on something not a string  **]  ", which has entries 0 up to 31 **]

**Comment by MarcCote** [comment]
It seems that with the game Anchorhead (`anchor.z8`), there's one point that can't be obtained. The main reason being frotz not dealing well with reading the Historical book where you have to navigate through the sections of a book using command line menu.

To reproduce run this command line:
> (echo -e "\n\n\nSe\nPush can against wall\nClimb can\nUp\nWest\nPush play\nEast\nLook up Verlac\nWest\nUnlock Door\nWest\nWest\nWest\nNW\nWest\nRead book\nShow keys to Michael\nEast\nSE\nEast\nSouth\nSouth\nEast\nSouth\nSW\nNW\nUnlock door\nNorth\nClose door\nLock door\nUp\nNorth\nUndress\nDrop all\nWest\nBathe\nEast\nLie in bed\nSleep\nLook\nwait\nwait\nGet wallet\nOpen wallet\nGet card\nLeave bed\nDress\nGet all\nWear coat\nSouth\nDown\nWest\nOpen cupboard\nPull lining\nRead journal\nNorth\nGet flashlight\nOpen Cabinet\nGet matches\nTurn on flashlight\nNW\nGet broom\nUnlock door\nDown\nSouth\nSearch crates\nGet box\nDrop cardboardbox\nWipe web\nDrop broom\nGet key\nPut key on keyring\nNorth\nUp\nSE\nEast\nEast\nLook at paintings\nLook at scene\nSouth\nGet album\nLook up Wilhelm in album\nLook up Eustacia in album\nLook up Croseus in album\nWest\nUp\nEast\nNorth\nGet volume\nLook at bookshelf\nGet Poe\nExamine safe\nturn dial to 47\nturn dial to 11\nturn dial to 3\nLook in safe\nGet puzzle box and flute\nSouth\nSouth\nOpen jewelry box\nGet silver locket\nPush bed\nLook in hole\nGet pages\nRead pages\nNorth\nWest\nDown\nNorth\nWest\nUnlock door\nNorth\nNW\nUnlock crypt\nDown\nExamine coffin\nLook up William on nameplate\nOpen William coffin\nGet skull\nUp\nSE\nSouth\nClose door\nLock door\nEast\nSouth\nUnlock door\nSouth\nSE\nNE\nNorth\nWest\nGet newspaper\nRead newspaper\nSouth\nSouth\nSE\nLook up Edward in record\nLook up Mordecai in record\nLook up Elijah inrecord\nLook up Heinrich in record\nLook up Wilhelm in record\nNW\nSW\nLook up Edward in record\nLook up Mordecai in Record\nLook up Elijah inrecord\nLook up Heinrich in record\n Look up Wilhelm in record\nNE\nNorth\nNorth\nNorth\nNorth\nWest\nNW\nWest\nNorth\nRing bell\nShow card to librarian\nAsk librarian for book\nOpen Historical\nGet slip of paper\nRead slip of paper\nRead historical" && cat) | dfrotz -h 999 -s 0 roms/anchor.z8

**Comment by MarcCote** [comment]
# Sorcerer
It seems after ~50 moves into the walkthrough, the player needs to refer to their [manual](http://infodoc.plover.net/manuals/temp/sorcerer.pdf) to get the right sequence of buttons to press.

**Comment by mhauskn** [comment]
Hmm, that sounds quite difficult for learning agents. Is it still possible to encode a walkthrough with a random seed that will work?

**Comment by mhauskn** [comment]
Regarding the Anchorhead issue - is this something we should submit to Frotz's repo for an upstream fix?

**Comment by MarcCote** [comment]
@mhauskn I do have a working walkthrough for Sorcerer when seed=0. I don't know, if issuing extra actions changes the correct sequence of buttons, though (i.e. is the sequence determine when the game starts or later in the game).

For anchorhead, maybe we should create an issue on the Frotz's repo. I want to investigate it more though.

**Comment by mhauskn** [comment]
Yeah, cutthroats has been problematic in a variety of ways. And it so far has not been included in published results. I would consider cutting it from the list of supported games.

Regarding Sorcerer - I think having a walkthrough when seed=0 sounds fine to me.

Regarding Anchorhead - please do create the issue. I tried navigating around the book in dfrotz but couldn't seem to figure out how to get the 1pt of score.

**Comment by MarcCote** [comment]
I did find a working walkhrough for cutthroats. But yeah, I don't think it would be feasible to finish with external knowledge.

**Comment by MarcCote** [comment]
By the way, I didn't raise an issue with frotz about anchorhead since I managed to read through the section of the book (using 'next' for changing chapter, and 'quit' for quitting the book). However, I didn't get the point. Maybe the [tutorial](http://mirror.ifarchive.org/if-archive/solutions/AnchorSolve.txt) I followed was wrong.

**Comment by MarcCote** [comment]
I believe the following games are not problematic anymore as we are flagging `showverb` as an illegal commands.
Game | Command | Error Type 
------------ | ------------- | -------------
detective.z5 | showverb eight | __Hang__ [** Programming error: tried to read from -->1115 in the array " [** Programming error: tried to print (string) on something not a string  **]  ", which has entries 0 up to 31 **]
omniquest.z5 | showverb one | __Hang__ [** Programming error: tried to read from -->914 in the array " [** Programming error: tried to print (string) on something not a string  **]  ", which has entries 0 up to 31 **]
